common-app-lib/src/cmr/common_app/services/kms_lookup.clj

```diff
@@ -280,95 +296,122 @@
-        ;; to the UMM Platform Type field.  This will avoid complications with facets.
-        umm-c-keyword (set/rename-keys umm-c-keyword {:type :category})
-        comparison-map (normalize-for-lookup umm-c-keyword (kms-scheme->fields-for-umm-c-lookup
-                                                            keyword-scheme))
-        umm-c-cache (hash-cache/context->cache context kms-umm-c-cache-key)
-        [tm value] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))
-        _ (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-platforms" kms-umm-c-cache-key tm)
-        value (if value
-                value
-                (let [_ (load-cache-if-necessary context umm-c-cache kms-umm-c-cache-key)
-                      [tm vlue] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))]
-                  (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-platforms" kms-umm-c-cache-key tm)
-                  vlue))]
-    (if (get umm-c-keyword :long-name)
-      ;; Check both longname and shortname
-      (get-in value [comparison-map])
-      ;; Check just shortname
-      (-> value
-          (remove-long-name-from-kms-index)
-          (get comparison-map)))))
+  (try
+    (let [umm-c-keyword (csk-extras/transform-keys csk/->kebab-case umm-c-keyword)
+          ;; CMR-3696 This is needed to compare the keyword category, which is mapped
+          ;; to the UMM Platform Type field.  This will avoid complications with facets.
+          umm-c-keyword (set/rename-keys umm-c-keyword {:type :category})
+          comparison-map (normalize-for-lookup umm-c-keyword (kms-scheme->fields-for-umm-c-lookup keyword-scheme))
+          umm-c-cache (hash-cache/context->cache context kms-umm-c-cache-key)
+          [tm value] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))
+          _ (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-platforms" kms-umm-c-cache-key tm)
+          value (if value
+                  value
+                  (let [_ (load-cache-if-necessary context umm-c-cache kms-umm-c-cache-key)
+                        [tm vlue] (util/time-execution (hash-cache/get-value umm-c-cache kms-umm-c-cache-key keyword-scheme))]
+                    (rl-util/log-redis-read-complete "lookup-by-umm-c-keyword-platforms" kms-umm-c-cache-key tm)
+                    vlue))]
+      (if (get umm-c-keyword :long-name)
+        ;; Check both longname and shortname
+        (get-in value [comparison-map])
+        ;; Check just shortname
+        (-> value
+            (remove-long-name-from-kms-index)
+            (get comparison-map))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "lookup-by-umm-c-keyword-platforms found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.